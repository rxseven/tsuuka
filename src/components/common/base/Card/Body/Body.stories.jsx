import React from 'react';
import { storiesOf } from '@storybook/react';

import notes from 'stories/shared/notes';
import Body from './index';

const title = 'Components/Common/Base/Card/Body';
const component = <Body>Card Body</Body>;

storiesOf(title, module).add('default', () => component, {
  notes: notes.minimal
});
