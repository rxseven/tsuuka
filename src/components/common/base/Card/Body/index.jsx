import PropTypes from 'prop-types';
import exact from 'prop-types-exact';
import React from 'react';

const propTypes = exact({
  children: PropTypes.node.isRequired
});

function Body({ children }) {
  return <div className="card-body">{children}</div>;
}

Body.propTypes = propTypes;

export default Body;
