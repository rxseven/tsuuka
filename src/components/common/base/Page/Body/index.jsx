import PropTypes from 'prop-types';
import exact from 'prop-types-exact';

const propTypes = exact({
  children: PropTypes.node.isRequired
});

function Body({ children }) {
  return children;
}

Body.propTypes = propTypes;

export default Body;
