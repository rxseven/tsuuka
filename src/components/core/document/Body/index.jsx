import PropTypes from 'prop-types';
import exact from 'prop-types-exact';
import React from 'react';
import styled from 'styled-components';

import HTML from 'constants/elements/html';

const Frame = styled.div.attrs({ id: HTML.body })`
  flex-basis: auto;
  flex-grow: 1;
  flex-shrink: 0;
`;

const propTypes = exact({
  children: PropTypes.element.isRequired
});

function Body({ children }) {
  return <Frame>{children}</Frame>;
}

Body.propTypes = propTypes;

export default Body;
