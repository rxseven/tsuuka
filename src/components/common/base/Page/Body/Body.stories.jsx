import React from 'react';
import { storiesOf } from '@storybook/react';

import mock from 'stories/mock';
import notes from 'stories/shared/notes';
import Body from './index';

const { text } = mock;
const title = 'Components/Common/Base/Page/Body';
const component = <Body>{text}</Body>;

storiesOf(title, module).add('default', () => component, {
  notes: notes.minimal
});
