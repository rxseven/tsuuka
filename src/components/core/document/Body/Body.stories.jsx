import React from 'react';
import { storiesOf } from '@storybook/react';

import mock from 'stories/mock';
import notes from 'stories/shared/notes';
import Body from './index';

const { content } = mock.elements;
const title = 'Components/Core/Document/Body';
const component = <Body>{content}</Body>;

storiesOf(title, module).add('default', () => component, {
  notes: notes.minimal
});
