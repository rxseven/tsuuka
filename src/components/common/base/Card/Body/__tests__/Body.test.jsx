import { shallow } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';

import Body from '../index';

describe('<Card.Body />', () => {
  // Arrange
  const props = { children: 'content' };
  const component = <Body {...props} />;

  describe('Unit tests', () => {
    it('should render without crashing', () => {
      // Act
      const wrapper = shallow(component);

      // Assert
      expect(wrapper).toBeDefined();
    });
  });

  describe('Snapshot tests', () => {
    it('should render correctly', () => {
      // Act
      const tree = renderer.create(component).toJSON();

      // Assert
      expect(tree).toMatchSnapshot();
    });
  });
});
