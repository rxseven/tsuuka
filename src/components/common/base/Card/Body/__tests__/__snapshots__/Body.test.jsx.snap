// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card.Body /> Snapshot tests should render correctly 1`] = `
<div
  className="card-body"
>
  content
</div>
`;
